Tests/Functional/DiscoveredClientsTest.php

```diff
@@ -71,9 +71,32 @@ public function testDiscoveredAsyncClientWithProfilingEnabled()
         $this->assertEquals('auto_discovered_async', NSA::getProperty($plugins[0], 'client'));
     }
 
-    private function getContainer($debug)
+    public function testDisabledDiscovery()
     {
-        static::bootKernel(['debug' => $debug]);
+        $container = $this->getContainer(true, 'discovery_disabled');
+
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_client'));
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_async'));
+        $this->assertFalse($container->has('httplug.strategy'));
+    }
+
+    public function testForcedDiscovery()
+    {
+        $container = $this->getContainer(true, 'discovery_forced');
+
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_client'));
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_async'));
```

Comment:
Should these be false on both `testDisabledDiscovery` and `testForcedDiscovery`?